File: Modeling/app/src/main/java/pisarev/com/modeling/mvp/model/Point2D.java
package pisarev.com.modeling.mvp.model;

public class Point2D {

    public float getX() {
        return x;
    }

    public void setX(float x) {
        this.x = x;
    }

    public float getZ() {
        return z;
    }

    public void setZ(float z) {
        this.z = z;
    }

    private float x;
    private float z;

    public Point2D(float x, float z) {
        this.x=x;
        this.z=z;
    }

    public float distance(float x, float z) {
        float a = getX() - x;
        float b = getZ() - z;
        return (float) Math.sqrt(a * a + b * b);
    }

    public float angle(float x, float y) {
        final float ax = getX();
        final float ay = getZ();

        final float delta = (float) ((ax * x + ay * y) / Math.sqrt(
                        (ax * ax + ay * ay) * (x * x + y * y)));

        if (delta > 1.0) {
            return 0.0f;
        }
        if (delta < -1.0) {
            return 180.0f;
        }

        return (float) Math.toDegrees(Math.acos(delta));
    }
}

File: Modeling/app/src/main/java/pisarev/com/modeling/mvp/contur/Point2D.kt
package pisarev.com.modeling.mvp.contur

import kotlin.math.acos
import kotlin.math.sqrt

class Point2D {

    var x: Double
    var z: Double

    constructor(x: Float, z: Float) {
        this.x = x.toDouble()
        this.z = z.toDouble()
    }

    fun distance(x: Float, z: Float): Float {
        val a = x - x
        val b = z - z
        return sqrt((a * a + b * b).toDouble()).toFloat()
    }

    fun angle(x: Float, y: Float): Float {
        val ax = x
        val ay = z
        val delta = ((ax * x + ay * y) / sqrt(
            (
                    (ax * ax + ay * ay) * (x * x + y * y))
        )).toFloat()
        if (delta > 1.0) {
            return 0.0F
        }
        return if (delta < -1.0) {
            180.0F
        } else Math.toDegrees(acos(delta.toDouble())).toFloat()
    }

}